packages/functions/src/index.ts

```diff
@@ -3,8 +3,53 @@ import * as functions from "firebase-functions";
 import {checkAuthToken} from "./middlewares/check-auth-token";
 import authRouter from "./resources/auth/auth.router";
 import userRouter from "./resources/user/user.router";
+import * as admin from "firebase-admin";
+import {Change, onDocumentCreated,
+  onDocumentDeleted,
+  onDocumentUpdated,
+  QueryDocumentSnapshot} from "firebase-functions/v2/firestore";
+import {updatePublishedEvent}
+  from "./resources/bot/actions/updatePublishedEvent";
+import {sendToChannel} from "./resources/bot/actions/sendToChannel";
+import {deleteTelegramMessage}
+  from "./resources/bot/actions/deleteTelegramMessage";
+
+admin.initializeApp();
+
+export const publishToTelegram =
+onDocumentCreated("events/{eventId}", (event) => {
+  const snapshot = event.data;
```

Comment:
let's log input validation and error handling here and other functions (updatePublishedEventTrigger):
1) Check that expected parameter is present and log error:
```
import { logger } from "firebase-functions";
...
      if (!snapshot) {
        logger.error("No data associated with the event");
        return;
      }
```

2) Wait for function execution, catch possible errors and check result
```
try {
    const data = snapshot.data();
    await sendToChannel(data);
} catch (err) {
   logger.error('Unable to send post to channel", err);
}
```